lib/errors.js

```diff
@@ -1,39 +1,73 @@
+'use strict';
+
+var util = require('util');
+
+/**
+ * The base class for all thinky errors
+ *
+ * @constructor
+ * @alias Error
+ */
+var ThinkyError = function(message) {
+  this.name = 'ThinkyError';
+  this.message = message || 'ThinkyError';
+  Error.captureStackTrace && Error.captureStackTrace(this, this.constructor);
+};
+util.inherits(ThinkyError, Error);
+
 /**
  * DocumentNotFound error which is returned when `get` returns `null`.
  * "Extends" Error
  */
-function DocumentNotFound(message) {
-  Error.captureStackTrace(this, DocumentNotFound);
+var DocumentNotFoundError = function(message) {
```

Comment:
Hum, people currently access this error with `thinky.Errors.DocumentNotFound`. While I'm not really opposed to `thinky.Errors.DocumentNotFound`, there is not a reason big enough to break compatibility here.
Could you switch it back to ``DocumentNotFound`?